app/views/messages/index.haml

```diff
@@ -3,7 +3,7 @@
     %thead
       = render partial: @view.dup
     %tbody
-      - if @messages.empty?
+      - if @messages.reject { |m| m.blocked?(current_user) }.empty?
```

Comment:
Maybe filter this on the model? Avoids having to do the each/next too. Or is that intended to save memory? And then you can also test whether it correctly filters, which seems useful